contrib/generic_proxy/filters/network/source/codecs/kafka/config.cc

```diff
@@ -0,0 +1,92 @@
+#include "contrib/generic_proxy/filters/network/source/codecs/kafka/config.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace GenericProxy {
+namespace Codec {
+namespace Kafka {
+
+KafkaServerCodec::KafkaServerCodec() : response_encoder_(response_buffer_) {}
+
+void KafkaServerCodec::setCodecCallbacks(GenericProxy::ServerCodecCallbacks& callbacks) {
+  request_callbacks_ = std::make_shared<KafkaRequestCallbacks>(callbacks);
+  request_decoder_ = std::make_shared<NetworkFilters::Kafka::RequestDecoder>(
+      std::vector<NetworkFilters::Kafka::RequestCallbackSharedPtr>{request_callbacks_});
+}
+
+void KafkaServerCodec::decode(Envoy::Buffer::Instance& buffer, bool) {
+  request_buffer_.move(buffer);
```

Comment:
can we use the original buffer?

Reply:
Yeah. But when we doing the decoding, the callbacks will be called and I am afraid that the original buffer maybe changed by the callbacks in some ways. (Although it's almost impossible, but moving also very cheap.)

If you think this is a block point, I can change it.

Reply:
I see it, I think it's fine.